utils/ipfs.ts

```diff
@@ -0,0 +1,12 @@
+import base58 from 'bs58';
+
+export function byte32ToIPFSCIDV0(hexstr: string): string {
+  /**
+   * Convert 32 bytes hex string to ipfscidv0.
+   * @param hexstr - 32 Bytes long string
+   * @returns IPFS CID Version 0
+   */
+  const binaryStr = Buffer.from(hexstr, 'hex');
```

Comment:
Are we sure we will always get a correct encoded and value for the image hash from Omen?

In my experience using buffers, they can become tricky with edge cases. I saw you're checking for the image hash to exist as a string, just pondering on the formats and encoding

Reply:
Yes, we will always get a good value. Otherwise the image will fails, which isn't a big deal.